docs/bos/queryapi/context.md

```diff
@@ -0,0 +1,348 @@
+  block_timestamp: blockTimeStampA
+};
+
+
+const insertPostDataB = {
+  id: postId
+  account_id: accountIdB,
+  block_height: blockHeightB,
+  block_timestamp: blockTimeStampB
+};
+// Insert new post to Posts table
+await context.db.Posts.upsert(
+  [ insertPostDataA, insertPostDataB ], 
+  [‘account_id’, ‘id’], 
+  [‘block_height’, ‘block_timestamp’);
+```
+
+In this example, two objects are being inserted. However, if a row already exists in the table where the `account_id` and `id` are the same, then `block_height` and `block_timestamp` would be overwritten in those rows to the value in the object in the call which is conflicting. 
+
+### Delete
+
+This method deletes all objects in the row that match the object passed in. Caution should be taken when using this method. It currently only support **AND** and exact match, just like in the [select method](#select). That doubles as a safety measure against accidentally deleting a bunch of data. All deleted rows are returned so you can always insert them back if you get back more rows than expected. (Or reindex your indexer if needed) 
+
+#### Input
+
+```js
+await context.db.TableName.delete(whereObj)
+```
+
+As stated, only a single object is allowed. 
+
+#### Example
+
+```js
+await context.db.delete_post_likes(
+  { 
+    account_id: likeAuthorAccountId, 
+    post_id: postId
+  }
+);
+```
+
+In this example, any rows where `account_id` and `post_id` match the supplied value are deleted. All deleted rows are returned. 
+
+### Auto Complete
+
+:::tip
+Autocomplete works while writing the schema and before publishing to the chain. In other words, you don't need to publish the indexer to get autocomplete.
+:::
+
+As mentioned, the [DB methods](#db-methods) are generated when the schema is read. 
+In addition to that, TypeScript types are generated which represent the table itself. These types are set as the parameter types. This provides autocomplete and strong typing in the IDE. These restrictions are not enforced on the runner side and are instead mainly as a suggestion to help guide the developer to use the methods in a way that is deemed correct by QueryAPI.
+
+
+:::info Types
+
+You can also generate types manually. Clicking the `<>` button generates the types. It can be useful for debugging and iterative development while writing the schema.
+
+![auto complete](/docs/queryapi/autocomp-types.png)
+
+:::
+
+#### Compatibility
+
+By default, current indexers have a `context` object included as a parameter in the top-level `async function getBlock`. This prevents autocomplete, as the local `context` object shadows the global one, preventing access to it. Users need to manually remove the `context` parameter from their indexers to get the autocomplete feature. For example:
+
+```js
+async function getBlock(block: Block, context) {
+```
+
+should become
+
+```js
+async function getBlock(block: Block) {
+```
+
+#### Examples
+
+Some screenshots of autocomplete and strong typing in action: 
```

Comment:
Maybe rephrase this to something more direct? Like: "Here are some screenshots which demonstrate autocomplete on methods, strong typing, and field names."

Reply:
fixed